go/src/koding/kites/kloud/provider/marathon/schema.go

```diff
@@ -0,0 +1,96 @@
+package marathon
+
+import (
+	"errors"
+	"net/http"
+	"net/url"
+	"time"
+
+	"koding/kites/kloud/stack"
+	"koding/kites/kloud/stack/provider"
+
+	marathon "github.com/gambol99/go-marathon"
+)
+
+var schema = &provider.Schema{
+	NewCredential: func() interface{} {
+		return &Credential{}
+	},
+	NewBootstrap: nil,
+	NewMetadata: func(m *stack.Machine) interface{} {
+		if m == nil {
+			return &Metadata{}
+		}
+
+		meta := &Metadata{}
+
+		return meta
+	},
+}
+
+var (
+	_ stack.Validator = (*Credential)(nil)
+	_ stack.Validator = (*Metadata)(nil)
+)
+
+// Credential represents credential information
+// that are required to deploy a Marathon app.
+type Credential struct {
+	URL                string `json:"url" bson:"url" hcl:"url"`
+	BasicAuthUser      string `json:"basic_auth_user" bson:"basic_auth_user" hcl:"basic_auth_user"`
+	BasicAuthPassowrd  string `json:"basic_auth_password" bson:"basic_auth_password" hcl:"basic_auth_password"`
+	RequestTimeout     int    `json:"request_timeout" bson:"request_timeout" hcl:"request_timeout"`
+	DeploymentTimepout int    `json:"deployment_timeout" bson:"deployment_timeout" hcl:"deployment_timeout"`
```

Comment:
ok got it for json since we decode/encode them multiple times, but it should not be the case for bson, it is only valid for mongodb. Anyhow for the sake of consistency with other providers, we can leave it as is.